### ISSUE TITLE
have the docker version report if it's out of date

### ISSUE THREAD

**Issue by trombonekenny** [PrairieLearn/PrairieLearn#885]
Maybe it could check at the first screen or when loading from disk?
Check against GitHub, or prairielearn.engr? Or something with hub.docker?

The error message could include instructions on how to `docker pull` an update.

**Comment by trombonekenny** [comment]
@nwalters512 This came up again in office hours this week. Turns out that's an old feature request. :)

**Comment by mwest1066** [comment]
Actually checking this sounds like a nuisance. Maybe we could just print a message when the container starts up like `run "docker pull prairielearn/prairielearn" to get the latest version`

**Comment by nwalters512** [comment]
Agreed that actually checking this would not be fun. The fact that we now publish region- specific tags (e.g. `prairielearn/prairielearn:us-prod-live` vs `prairielearn/prairielearn:ca-live`) that users might be using makes this more complicated, as we can't really give precise instructions.

https://github.com/PrairieLearn/PrairieLearn/pull/7082 added documentation for these new tags and recommends the use of `--pull=always`. @jonatanschroeder what do you think about recommending the use of `--pull=always` for our standard Docker instructions, not just those for environment-specific revisions? Would that mitigate this issue enough to close it?

**Comment by jonatanschroeder** [comment]
> #7082 added documentation for these new tags and recommends the use of `--pull=always`. @jonatanschroeder what do you think about recommending the use of `--pull=always` for our standard Docker instructions, not just those for environment-specific revisions? Would that mitigate this issue enough to close it?

A couple of issues that may come up:
* anyone that keeps long-running Docker containers in the background would not be pulled until the container is restarted, though I'm not sure if there is an easy way to handle that case.
* anyone that already has their environment set up with a specific command is unlikely to switch to the new command unless explicitly instructed to do so. Same comment as above.
* pulling the container for prairielearn/prairielearn is actually a pretty massive download, and pretty inconsequential in most common cases. This is a lesser deal in *-live tags, since they are not updated as often, but personally I usually only pull the PL image in the background and rely on `make deps` and `git pull` to take care of the rest.

I think it may make sense to move the `*-live` documentation as the default instead of the `latest` for the majority of users. In that case the instructions using the latest version without pull=always would typically be used by those that are doing so on purpose, which presumably means they acknowledge the implications.
